.github/workflows/publish.yml

```diff
@@ -1,21 +1,55 @@
 name: "publish"
-on: workflow_dispatch
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - "*"
 
 jobs:
   publish-tauri:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-20.04
-          - macos-11
-          - windows-2022
+        build:
+          # for ubuntu with arch, we need more testing with cross compilation libraries
+          # since releasing on arm-linux is not a priority, we skipped it for now
+          - os: ubuntu-20.04
+            target: x86_64-unknown-linux-gnu
+            suffix: ubuntu-x86_64
+            rustflags: ""
+          - os: macos-12
+            target: x86_64-apple-darwin
+            suffix: macos-x86_64
+            rustflags: ""
+          - os: macos-12
+            target: aarch64-apple-darwin
+            suffix: macos-aarch64
+            rustflags: ""
+          - os: windows-2022
+            target: x86_64-pc-windows-msvc
+            suffix: windows-x86_64
+            rustflags: ""
+
+    runs-on: ${{ matrix.build.os }}
+
+    env:
+      PRODUCTION_TARGET: src-tauri/target/${{ matrix.build.target }}/release/bundle
+      RUSTFLAGS: ${{ matrix.build.rustflags }}
 
-    runs-on: ${{ matrix.os }}
     steps:
+      - name: Maximize build space
+        if: runner.os == 'Linux'
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf /opt/ghc
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
```

Comment:
Are these still necessary? I'm worried this is just a temporary solution.

Fine for now, but we need to find longer-term solution somehow.

Reply:
Hmm, I thought since we are not planning to use dotnet and ghc, it could be a permanent solution...
I think it is still necessary, I can double-check in a fork, and look for other solutions if you like

Reply:
Well, yes, but it only frees some space for us. If we fit into unmodified I'd prefer to not delete anything, but not blocking of course.